src/Raven.Server/RavenServer.cs

```diff
@@ -237,6 +244,62 @@ void ConfigureKestrel(KestrelServerOptions options)
             }
         }
 
+        private void AssertServerCanContactItselfWhenAuthIsOn()
+        {
+            var url = Configuration.Core.PublicServerUrl.HasValue ? Configuration.Core.PublicServerUrl.Value.UriValue : Configuration.Core.ServerUrls[0];
+
+            try
+            {
+                using (var httpMessageHandler = new HttpClientHandler())
+                {
+                    httpMessageHandler.SslProtocols = SslProtocols.Tls12;
+
+                    // Using the server certificate as a client certificate to test if we can talk to ourselves
+                    httpMessageHandler.ClientCertificates.Add(Certificate.Certificate);
+                    using (var client = new HttpClient(httpMessageHandler)
+                    {
+                        BaseAddress = new Uri(url)
+                    })
+                    {
+                        var response = client.GetAsync("/setup/alive").Result;
```

Comment:
In some cases, this can hang for a LOT of time (minutes).
We need to write something to console about this and the log.
So user have a way to understand what is going on.

Also, does this _have_ to be syncronous?